src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -288,20 +303,182 @@ private static ProjectItemGroupElement CreateItemGroup(Project project, string f
             }
         }
 
+        /// <summary>
+        /// A simple check for some of the evaluated properties to check
+        /// if the project is package reference project or not
+        /// </summary>
+        /// <param name="project"></param>
+        /// <returns></returns>
+        internal static bool IsPackageReferenceProject(Project project)
+        {
+            return (project.GetPropertyValue(RESTORE_STYLE_TAG) != "" ||
+                    project.GetPropertyValue(PACKAGE_REFERENCE_TYPE_TAG) != "" ||
+                    project.GetPropertyValue(NUGET_STYLE_TAG) != "" ||
+                    project.GetPropertyValue(ASSETS_FILE_PATH_TAG) != "");
+        }
+
+        /// <summary>
+        /// Prepares the dictionary that maps frameworks to packages top-level
+        /// and transitive.
+        /// </summary>
+        /// <param name="userInputFrameworks">A list of framework names</param>
+        /// <param name="assetsFile">Assets file for all targets and libraries</param>
+        /// <param name="transitive">Include transitive packages in the result</param>
+        /// <returns></returns>
+        internal Dictionary<string, Tuple<IEnumerable<PRPackage>, IEnumerable<PRPackage>>> GetResolvedVersions(
+            IList<string> userInputFrameworks,LockFile assetsFile, bool transitive)
+        {
+            var resultPackages = new Dictionary<string, Tuple<IEnumerable<PRPackage>, IEnumerable<PRPackage>>>();
+
+            var foundTargetFrameworks = new List<TargetFrameworkInformation>();
+            if (userInputFrameworks.Count() == 0)
+            {
+                foundTargetFrameworks = assetsFile.PackageSpec.TargetFrameworks.ToList();
+            }
+            else
+            {
+                foundTargetFrameworks.AddRange(assetsFile.PackageSpec.TargetFrameworks.Where(tfm => userInputFrameworks.Contains(tfm.ToString())));
+            }
+
+            foreach (var target in assetsFile.Targets)
+            {
+                var tfmInformation = foundTargetFrameworks.Where(tfm => assetsFile.GetTarget(tfm.FrameworkName, null).Equals(target));
+
+                //If the target is not in the list of target frameworks, skip the target
+                if (!tfmInformation.Any())
+                {
+                    continue;
+                }
+
+                //The packages for the framework that were retrieved with GetRequestedVersions
+                var frameworkDependencies = tfmInformation.Single().Dependencies;
```

Comment:
There shouldn't be as I understand, but just in case, I switched it to `FirstOrDefault`

Reply:
If you use `FirstOrDefault` is because you expect it can be `null`, if this is correct make sure to do the appropriate null checks. If not, you can use `First`